sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -375,14 +370,20 @@ static int _find_resource(coap_pkt_t *pdu, coap_resource_t **resource_ptr,
 
     /* Find path for CoAP msg among listener resources and execute callback. */
     gcoap_listener_t *listener = _coap_state.listeners;
+
+    uint8_t uri[NANOCOAP_URI_MAX];
+    if (coap_get_uri(pdu, uri) <= 0) {
+        return GCOAP_RESOURCE_NO_PATH;
+    }
+
     while (listener) {
         coap_resource_t *resource = listener->resources;
         for (size_t i = 0; i < listener->resources_len; i++) {
             if (i) {
                 resource++;
             }
 
-            int res = strcmp((char *)&pdu->url[0], resource->path);
+            int res = strcmp((char *)&uri[0], resource->path);
```

Comment:
coap_get_uri() inserts the terminator, but I agree it's better to be safe.

Reply:
Small note here, I'm not sure whether strncmp is in the C99 libc. (It's not the case with strnlen by default)

Reply:
Use of strncmp() adds 40 bytes to the samr21-xpro executable text space. Since coap_get_uri() documentation asserts that it inserts the terminator, would you be OK with beefing up the unit tests for that function as an alternative way to accomplish the goal?

Reply:
Agreed, especially since it is a string build in our own functions.

Reply:
Merged in tests in #9444.